bin/truffleruby

```diff
@@ -39,44 +27,74 @@ while [ -h "$SELF_PATH" ]; do
     SYM=$(readlink "$SELF_PATH")
     SELF_PATH=$(cd "$DIR" && cd "$(dirname "$SYM")" && pwd)/$(basename "$SYM")
 done
+
+# this is graalvm/language/ruby on graalvm
 root=$(dirname "$(dirname "$SELF_PATH")")
 
-mx_jar="$root/mxbuild/dists/ruby.jar"
-mvn_jar="$root/lib/truffleruby.jar"
-mx_time=$(mtime "$mx_jar")
-mvn_time=$(mtime "$mvn_jar")
-
-if [ -z "$JAVACMD" ]; then
-  if [ -z "$JAVA_HOME" ]; then
-    JAVACMD='java'
-  else
-    JAVACMD="$JAVA_HOME/bin/java"
-  fi
+root_parent=$(dirname "$root")
+if [ $(basename "$root") = ruby ] && [ $(basename "$root_parent") = language ]; then
+    on_graalvm=true
+else
+    on_graalvm=false
+fi
```

Comment:
One way to make this less hacky would be to have `graalvm/bin/ruby` define an extra variable `$GRAALVM` or so, and then `source` this script. It does mean duplicating the symlink expansion though.